test/essentials.js

```diff
@@ -71,3 +71,64 @@ test('cloneNode', function (t) {
   t.equal(cloneParentEl.childNodes[1].type, el.nodeName)
   t.equal(cloneParentEl.childNodes[2].nodeType, 3)
 })
+
+test('compareDocumentPosition', function (t) {
+  /*
+   * DOM-structure
+   * <documentEl>
+   *   <parentEl>
+   *     <siblingOne />
+   *     <siblingTwo />
+   *   </parentEl>
+   *   <otherEl>
+   *     <siblingThree />
+   *   </otherEl>
+   * </documentEl>
+   * ---
+   * <otherDocumentEl />
+   */
+  var documentEl = ReactFauxDOM.createElement('div')
+  var parentEl = ReactFauxDOM.createElement('div')
+  var siblingOne = ReactFauxDOM.createElement('div')
+  var siblingTwo = ReactFauxDOM.createElement('div')
+  var otherEl = ReactFauxDOM.createElement('div')
+  var siblingThree = ReactFauxDOM.createElement('div')
+  var otherDocumentEl = ReactFauxDOM.createElement('div')
+
+  documentEl.appendChild(parentEl)
+  documentEl.appendChild(otherEl)
+  parentEl.appendChild(siblingOne)
+  parentEl.appendChild(siblingTwo)
+  otherEl.appendChild(siblingThree)
+
+  t.plan(8)
+  t.equal(
+    siblingOne.compareDocumentPosition(siblingTwo),
+    Element.DOCUMENT_POSITION_FOLLOWING
+  )
+  t.equal(
+    siblingTwo.compareDocumentPosition(siblingOne),
+    Element.DOCUMENT_POSITION_PRECEDING
+  )
+  t.equal(
+    parentEl.compareDocumentPosition(siblingOne),
+    Element.DOCUMENT_POSITION_FOLLOWING + Element.DOCUMENT_POSITION_CONTAINED_BY
+  )
+  t.equal(
+    siblingTwo.compareDocumentPosition(siblingThree),
+    Element.DOCUMENT_POSITION_FOLLOWING
+  )
+  t.equal(
+    siblingTwo.compareDocumentPosition(parentEl),
+    Element.DOCUMENT_POSITION_PRECEDING + Element.DOCUMENT_POSITION_CONTAINS
```

Comment:
These tests look good, thanks for adding a lot! But once again, this should be a bitmask if I'm not mistaken.